crates/nu-cmd-base/src/arg_glob.rs

```diff
@@ -9,19 +11,33 @@ const FS_CMD_WALK_BEHAVIOR: WalkBehavior = WalkBehavior {
     link: LinkBehavior::ReadTarget,
 };
 
-// expand a glob into matching paths, using FS_CMD_WALK_BEHAVIOR
+// handle an argument that could be a literal path or a glob.
+// if literal path, return just that (whether user can access it or not).
+// if glob, expand into matching paths, using FS_CMD_WALK_BEHAVIOR
 //todo: see about returning WalkError or BuildError, or enabling miette and dealing with diagnostic
 pub fn arg_glob(
-    pattern: &Spanned<String>, // the glob
+    pattern: &Spanned<String>, // alleged path or glob
     include_dirs: bool,        // include dirs in results.  Default (f) only include files
     cwd: &PathBuf,             // current working directory
 ) -> Result<Vec<PathBuf>, ShellError> {
-    #[cfg(windows)]
-    let processed_item = &windows_pattern_hack(&pattern.item);
-    #[cfg(not(windows))]
-    let processed_item = &pattern.item;
+    // stat the path first, return path if not not found.
+    match fs::metadata(&pattern.item) {
```

Comment:
oh, nice! this is clever. I like it much better than the windows hack. nice idea.